File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/service/sed/helpers/StreamUtils.java
package no.nav.melosys.eessi.service.sed.helpers;

import java.util.Collection;
import java.util.stream.Stream;

import lombok.experimental.UtilityClass;

@UtilityClass
public class StreamUtils {
    public <T> Stream<T> nullableStream(Collection<T> collection) {
        return Stream.ofNullable(collection).flatMap(Collection::stream);
    }
}

File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/service/sed/helpers/StreamUtils.kt
package no.nav.melosys.eessi.service.sed.helpers

import java.util.stream.Stream

object StreamUtils {
    @JvmStatic
    fun <T> nullableStream(collection: Collection<T>?): Stream<T> {
        return collection?.stream() ?: Stream.empty()
    }
}
